### PR TITLE
Bump rabbitmq operator to v2.2.0

### PR DESCRIPTION
**Pull request by gibizer**
We cannot go higher as v2.3.0 pulls in new k8s and controller runtime versions we are not compatible with yet.

Also we cannot point to v2.2.0 directly as they are not properly releasing the v2 versions. See https://github.com/rabbitmq/cluster-operator/issues/1407

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -10,7 +10,9 @@ require (
 	github.com/openstack-k8s-operators/infra-operator/apis v0.0.0-20230720153501-076b82bb4427
 	github.com/openstack-k8s-operators/lib-common/modules/common v0.1.0
 	github.com/openstack-k8s-operators/lib-common/modules/test v0.1.1
-	github.com/rabbitmq/cluster-operator v1.14.0
+	// this is the commit pointed by v2.2.0 but we cannot import v2 directly
+	// as they are not properly releasing it
```
Comment (gibizer) [inline go.mod:14]: Mention https://github.com/rabbitmq/cluster-operator/issues/1407 here

**Comment by gibizer** [comment]
/test infra-operator-build-deploy 
prow did not re-run the job after the last push :/

**Comment by gibizer** [comment]
https://review.rdoproject.org/zuul/build/47697b5618a045688b828d4c5fd7a1b4/log/job-output.txt :
```
"oc wait pod -n openshift-nmstate -l component=kubernetes-nmstate-webhook --for condition=Ready --timeout=300s", "pod/nmstate-webhook-6c7cff4dd4-pcpg8 condition met", "WORKERS='' \\", "bash scripts/gen-nncp.sh", "Please set WORKERS"], "uid": 1000}
```

**Comment by gibizer** [comment]
/test infra-operator-build-deploy

I don't see the reason in the logs why the deploy failed. The OpenStackControlPlane CR is in Ready state.

**Comment by abays** [comment]
recheck

**Comment by gibizer** [comment]
We tagged v0.1.0 already so we can technically land this.

**Review by abays — APPROVED** [review]
/lgtm

**Pull request closed without merging**